core/src/main/java/com/novoda/noplayer/internal/exoplayer/SecurityRequirementBasedMediaCodecVideoRenderer.java

```diff
@@ -0,0 +1,73 @@
+package com.novoda.noplayer.internal.exoplayer;
+
+import android.content.Context;
+import android.os.Handler;
+
+import com.google.android.exoplayer2.Format;
+import com.google.android.exoplayer2.drm.DrmInitData;
+import com.google.android.exoplayer2.drm.DrmSessionManager;
+import com.google.android.exoplayer2.drm.FrameworkMediaCrypto;
+import com.google.android.exoplayer2.mediacodec.MediaCodecInfo;
+import com.google.android.exoplayer2.mediacodec.MediaCodecSelector;
+import com.google.android.exoplayer2.mediacodec.MediaCodecUtil;
+import com.google.android.exoplayer2.video.MediaCodecVideoRenderer;
+import com.google.android.exoplayer2.video.VideoRendererEventListener;
+
+import java.util.List;
+
+import androidx.annotation.Nullable;
+
+class SecurityRequirementBasedMediaCodecVideoRenderer extends MediaCodecVideoRenderer {
+
+    // Extension from MediaCodecVideoRenderer, we can't do anything about this.
+    @SuppressWarnings({"checkstyle:ParameterNumber", "PMD.ExcessiveParameterList"})
+    SecurityRequirementBasedMediaCodecVideoRenderer(Context context,
+                                                    MediaCodecSelector mediaCodecSelector,
+                                                    long allowedJoiningTimeMs,
+                                                    @Nullable DrmSessionManager<FrameworkMediaCrypto> drmSessionManager,
+                                                    boolean playClearSamplesWithoutKeys,
+                                                    boolean enableDecoderFallback,
+                                                    @Nullable Handler eventHandler,
+                                                    @Nullable VideoRendererEventListener eventListener,
+                                                    int maxDroppedFramesToNotify) {
+        super(
+                context,
+                mediaCodecSelector,
+                allowedJoiningTimeMs,
+                drmSessionManager,
+                playClearSamplesWithoutKeys,
+                enableDecoderFallback,
+                eventHandler,
+                eventListener,
+                maxDroppedFramesToNotify
+        );
+    }
+
+    @Override
+    protected int supportsFormat(MediaCodecSelector mediaCodecSelector,
```

Comment:
These two methods are the only usages I have seen of the `MediaCodecSelector`. I have not applied the same logic to the `AudioRenderers` so they will use whatever is dictated by this I guess.

Reply:
@zegnus do you think it is worth doing the same for the `audio renderer`?

Reply:
I think it is yes, now that we are on it let's get it sorted